src/index.ts

```diff
@@ -369,13 +369,49 @@ export type StorageOptions = ConstructorParameters<typeof BaseStorage>[0]
  * https://docs.plasmo.com/framework/storage
  */
 export class Storage extends BaseStorage {
-  get = async <T = string>(key: string) => {
+  get = async <T = string>(key: string|string[]) => {
+
+    if (Array.isArray(key)) {
+      const nsKeys = key.map((k) => ({
+        key: k,
+        namespaced: this.getNamespacedKey(k),
+      }));
+      const items = {};
+
+      await Promise.all(
+        nsKeys.map(async ({ namespaced, key }) => {
+          const rawData = await this.rawGet(namespaced);
+          items[key] = await this.parseValue(rawData);
+        })
+      );
+
+      return items;
+    }
+
     const nsKey = this.getNamespacedKey(key)
     const rawValue = await this.rawGet(nsKey)
     return this.parseValue(rawValue) as T
   }
 
-  set = async (key: string, rawValue: any) => {
+  set = async (key: string|Record<string, any>, rawValue?: any) => {
+    if (typeof key === 'object') {
+      const keyEntries = Object.entries(key);
+
+      await Promise.all(
+        keyEntries.map(async ([index, value]) => {
+          const nsKey = this.getNamespacedKey(index);
+          const jsonValue = JSON.stringify(value);
+          return this.rawSet(nsKey, jsonValue);
+        })
+      );
+
+      return null;
+    }
+
+    if (!rawValue) {
```

Comment:
this doesn't check for strict null, thus an empty string would throw the error as well.